### PR TITLE
Official Enjin Ruleset

### PR DESCRIPTION
**Pull request by Bradez**
Primary Domain: enjin.com

Additional Domains:
- enjincoin.io
- enjinwallet.io
- enjinx.io
- efinity.io

### CODE REVIEW

File: src/chrome/content/rules/Enjin.xml [hunk 1]
```diff
@@ -1,47 +1,64 @@
...
+
+	<rule from="^http://([\w-]+\.)?enjinx\.io/" to="https://$1enjinx.io/" />
+
+	<!-- Efinity -->
+	<target host="efinity.io" />
+	<target host="www.efinity.io" />
+
+	<rule from="^http://([\w-]+\.)?efinity\.io/" to="https://$1efinity.io/" />
+
+	<!-- CDN Rules -->
```
Comment (J0WI) [inline src/chrome/content/rules/Enjin.xml]: Please add a short comment, that they serve a mismatching cert.
Please also try to provide some `<test url="" />` for all of them.

File: src/chrome/content/rules/Enjin.xml [hunk 2]
```diff
@@ -1,47 +1,64 @@
...
+	(between the HTTP and HTTPS protocols) to occur.
 
-		- d2umfjgc9kwn9b.cloudfront.net
-			- assets-cloud.enjin.com
+	This functionality may be altered in the near future to ensure all sites use HTTPS and proxy
+	insecure material (such as scripts, stylesheets and images) through a HTTPS-enabled proxy.
 -->
-<ruleset name="Enjin" platform="mixedcontent" default_off='failed ruleset test'>
+
+<ruleset name="Enjin" platform="mixedcontent">
```
Comment (J0WI) [inline src/chrome/content/rules/Enjin.xml]: Are there any serious mixedcontent issues on the sites? I didn't found any so far.

**Comment by J0WI** [comment]
Thanks for providing this update! I hope you're able to switch to HTTPS soon, at best you also manage to get directly [preloaded](https://hstspreload.org/) in browsers.

File: src/chrome/content/rules/Enjin.xml (hunk 2 above)
Comment (Bradez) [reply src/chrome/content/rules/Enjin.xml]: The user's are capable of adding their own content which may cause mixed content, the official website has a flag to automatically upgrade all of our resources to HTTPS so you won't see mixed content on our main website.

File: src/chrome/content/rules/Enjin.xml (hunk 1 above)
Comment (Bradez) [reply src/chrome/content/rules/Enjin.xml]: This is a new website, owned by Enjin, it has no additional rules but the point is that the website will fully utilise HTTPS.

**Comment by Bradez** [comment]
Thank you for your comments, this is definitely something we're looking into although - given it's 9 years of work and we have many paying customers - it's not as simple to enforce HTTPS on all of the sites although we are very much seeking to do so in the upcoming future in which case we will definitely be utilising HSTS and preloading.

File: src/chrome/content/rules/Enjin.xml [hunk 3]
```diff
@@ -1,47 +1,64 @@
...
+	<target host="www.enjinwallet.io" />
+
+	<rule from="^http://([\w-]+\.)?enjinwallet\.io/" to="https://$1enjinwallet.io/" />
+
+	<!-- EnjinX -->
+	<target host="enjinx.io" />
+	<target host="www.enjinx.io" />
+	<target host="api.enjinx.io" />
+
+	<test url="http://api.enjinx.io/" />
```
Comment (Bradez) [inline src/chrome/content/rules/Enjin.xml]: I will remove this test URL for the time being as the target is intended to be a wildcard yet we only have 2 test URLs (www and api) so can't utilise the wildcard target. Should we ever change that, will be certain to ensure the target host is changed and test URLs added.

File: src/chrome/content/rules/Enjin.xml (hunk 1 above)
Comment (J0WI) [reply src/chrome/content/rules/Enjin.xml]: Just to clarify: this comment was referring to the CDN hosts below.
If your S3 bucket changed for some reason, test urls will ensure that we are aware that the location has changed.

Comment (Bradez) [reply src/chrome/content/rules/Enjin.xml]: I apologise, I missed that in my initial review of your comments, I'll add some test URLs for the CDN along with a comment. Thank you.

**Pull request merged**